flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/sinks/BatchSelectTableSink.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.sinks;
+
+import org.apache.flink.api.common.ExecutionConfig;
+import org.apache.flink.api.common.JobExecutionResult;
+import org.apache.flink.api.common.accumulators.SerializedListAccumulator;
+import org.apache.flink.api.common.typeutils.TypeSerializer;
+import org.apache.flink.api.java.Utils;
+import org.apache.flink.core.execution.JobClient;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.datastream.DataStreamSink;
+import org.apache.flink.table.api.TableException;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.api.internal.SelectTableSink;
+import org.apache.flink.table.runtime.types.TypeInfoDataTypeConverter;
+import org.apache.flink.table.sinks.StreamTableSink;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.AbstractID;
+import org.apache.flink.util.Preconditions;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * A {@link SelectTableSink} for batch select job.
+ *
+ * <p><strong>NOTES:</strong> This is a temporary solution,
+ * once FLINK-14807 is finished, the implementation should be changed.
+ */
+public class BatchSelectTableSink implements StreamTableSink<Row>, SelectTableSink {
+	private final TableSchema tableSchema;
+	private final String accumulatorName;
+	private final TypeSerializer<Row> typeSerializer;
+	private JobClient jobClient;
+
+	@SuppressWarnings("unchecked")
+	public BatchSelectTableSink(TableSchema tableSchema) {
+		this.tableSchema = SelectTableSinkSchemaConverter.convert(tableSchema);
+		this.accumulatorName = new AbstractID().toString();
+		this.typeSerializer = (TypeSerializer<Row>) TypeInfoDataTypeConverter
```

Comment:
why use this deprecated `TypeInfoDataTypeConverter`?

Reply:
There are no other options now, we must use this

Reply:
`TableSchema.toRowType()` has similar features but it's also a deprecated method and some types are not handled in `TableSchema.toRowType()`, like `BINARY`